From dcb24f5068f54f0fd3c30b691b8e3735973d008b Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Wed, 1 Apr 2009 16:15:27 +0200
Subject: [PATCH] load bt discover module only when installed

---
 src/daemon/default.pa.in |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/daemon/default.pa.in b/src/daemon/default.pa.in
index 57ccd9f..fa0683e 100755
--- a/src/daemon/default.pa.in
+++ b/src/daemon/default.pa.in
@@ -58,9 +58,9 @@ load-module module-detect
 .endif
 
 ### Automatically load driver modules for Bluetooth hardware
-#.ifexists module-bluetooth-discover@PA_SOEXT@
+.ifexists module-bluetooth-discover@PA_SOEXT@
 load-module module-bluetooth-discover
-#.endif
+.endif
 
 ### Load several protocols
 .ifexists module-esound-protocol-unix@PA_SOEXT@
-- 
1.6.2

