From 8f43f8ceae382fafcdf59533a5f3776b41c174cd Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Thu, 14 Jul 2016 17:35:54 +0200
Subject: [PATCH 02/18] tagstruct: don't forget to copy the length

---
 src/pulsecore/tagstruct.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/pulsecore/tagstruct.c b/src/pulsecore/tagstruct.c
index c29e49b..66ff8df 100644
--- a/src/pulsecore/tagstruct.c
+++ b/src/pulsecore/tagstruct.c
@@ -104,6 +104,7 @@ pa_tagstruct *pa_tagstruct_copy(pa_tagstruct*t) {
         tc = pa_xnew(pa_tagstruct, 1);
     tc->data = pa_xmemdup(t->data, t->length);
     tc->allocated = t->length;
+    tc->length = t->length;
     tc->rindex = 0;
     tc->type = PA_TAGSTRUCT_DYNAMIC;
 
-- 
2.9.3

