From c72a7db3b1e08b9ebef3a0f7da0b8e669f7e79fa Mon Sep 17 00:00:00 2001
From: Colin Guthrie <pulse@colin.guthr.ie>
Date: Wed, 8 Oct 2008 22:56:12 +0200
Subject: [PATCH] Fix a potential C++/C99 ism, add a log message on error condition

Signed-off-by: Lennart Poettering <lennart@poettering.net>
---
 src/pulsecore/pid.c |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/pulsecore/pid.c b/src/pulsecore/pid.c
index ce8ef19..99ba3e1 100644
--- a/src/pulsecore/pid.c
+++ b/src/pulsecore/pid.c
@@ -211,6 +211,7 @@ int pa_pid_file_create(const char *procname) {
     if ((pid = read_pid(fn, fd)) == (pid_t) -1)
         pa_log_warn("Corrupt PID file, overwriting.");
     else if (pid > 0) {
+        int ours = 1;
 
 #ifdef OS_IS_WIN32
         if ((process = OpenProcess(PROCESS_QUERY_INFORMATION, FALSE, pid)) != NULL) {
@@ -218,11 +219,13 @@ int pa_pid_file_create(const char *procname) {
 #else
         if (kill(pid, 0) >= 0 || errno != ESRCH) {
 #endif
-            int ours = 1;
 
             if (procname)
-                if ((ours = proc_name_ours(pid, procname)) < 0)
+                if ((ours = proc_name_ours(pid, procname)) < 0) {
+                    pa_log_warn("Could not check to see if pid %lu is a pulseaudio process. "
+                                "Asssuming it is and the daemon is already running.", (unsigned long) pid);
                     goto fail;
+                }
 
             if (ours) {
                 pa_log("Daemon already running.");
-- 
1.6.0.3

