From b041aac9b104008240cf4f3571de6a4637d94b1d Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 21 Oct 2008 18:40:01 +0200
Subject: [PATCH] include log.h near the end so that macro.h can be included in log.h and defines pa_bool_t properly

---
 src/pulsecore/macro.h |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/src/pulsecore/macro.h b/src/pulsecore/macro.h
index 2d031da..cf02696 100644
--- a/src/pulsecore/macro.h
+++ b/src/pulsecore/macro.h
@@ -30,7 +30,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-#include <pulsecore/log.h>
 #include <pulse/gccmacro.h>
 
 #ifndef PACKAGE
@@ -221,4 +220,7 @@ typedef int pa_bool_t;
 
 #endif
 
+/* We include this at the very last place */
+#include <pulsecore/log.h>
+
 #endif
-- 
1.6.0.3

