From 6fc7131ec231865d2c1ce8267013d9cf78f36e68 Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Fri, 15 Jul 2016 16:05:36 -0400
Subject: [PATCH 17/18] Make sure to set the pid in auth_cb

---
 src/modules/module-flatpak.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/modules/module-flatpak.c b/src/modules/module-flatpak.c
index cf428d4..9375a9b 100644
--- a/src/modules/module-flatpak.c
+++ b/src/modules/module-flatpak.c
@@ -510,7 +510,6 @@ static pa_hook_result_t client_put_cb(pa_core *c, pa_object *o, struct userdata
     pa_client *cl;
     uint32_t policy;
 
-pa_log("client put\n");
     pa_assert(c);
     pa_object_assert_ref(o);
 
@@ -523,6 +522,8 @@ pa_log("client put\n");
 
     client_data_new(u, cl->index, policy, cl->creds.pid);
 
+    pa_log("client put: policy %d, pid %u\n", policy, cl->creds.pid);
+
     return PA_HOOK_OK;
 }
 
@@ -543,6 +544,9 @@ static pa_hook_result_t client_auth_cb(pa_core *c, pa_object *o, struct userdata
 
     policy = find_policy_for_client(u, cl);
     cd->policy = policy;
+    cd->pid = cl->creds.pid;
+
+    pa_log("auth cb: policy %d, pid %u\n", cd->policy, cd->pid);
 
     return PA_HOOK_OK;
 }
-- 
2.9.3

