From 8e2a48a7a2baa49a900b9c42e8d08aca5db90d1f Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 21 Oct 2008 22:04:22 +0200
Subject: [PATCH] make the debug trap macro a proper macro in macro.h

---
 src/pulsecore/macro.h |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/src/pulsecore/macro.h b/src/pulsecore/macro.h
index cf02696..f9ce949 100644
--- a/src/pulsecore/macro.h
+++ b/src/pulsecore/macro.h
@@ -220,6 +220,12 @@ typedef int pa_bool_t;
 
 #endif
 
+#if defined(__i386__) || defined(__x86_64__)
+#define PA_DEBUG_TRAP __asm__("int $3")
+#else
+#define PA_DEBUG_TRAP raise(SIGTRAP)
+#endif
+
 /* We include this at the very last place */
 #include <pulsecore/log.h>
 
-- 
1.6.0.3

