diff -up pulseaudio-12.99.2/src/utils/qpaeq.qpaeq_python2 pulseaudio-12.99.2/src/utils/qpaeq
--- pulseaudio-12.99.2/src/utils/qpaeq.qpaeq_python2	2019-07-04 19:06:47.000000000 -0500
+++ pulseaudio-12.99.2/src/utils/qpaeq	2019-08-07 10:30:03.134944372 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 #    qpaeq is a equalizer interface for pulseaudio's equalizer sinks
 #    Copyright (C) 2009  Jason Newton <nevion@gmail.com
 #
@@ -18,7 +18,7 @@
 
 import os,math,sys
 try:
-    import PyQt5,sip
+#    import PyQt5,sip
     from PyQt5 import QtWidgets,QtCore
     import dbus.mainloop.pyqt5
     import dbus
