diff -up pulseaudio-11.1/src/pulsecore/memfd-wrappers.h.glibc_memfd pulseaudio-11.1/src/pulsecore/memfd-wrappers.h
--- pulseaudio-11.1/src/pulsecore/memfd-wrappers.h.glibc_memfd	2016-08-23 07:50:11.000000000 -0500
+++ pulseaudio-11.1/src/pulsecore/memfd-wrappers.h	2018-01-22 12:30:23.322574380 -0600
@@ -33,9 +33,9 @@
  * a whole conflicts with the original glibc header <fnctl.h>.
  */
 
-static inline int memfd_create(const char *name, unsigned int flags) {
+/* static inline int memfd_create(const char *name, unsigned int flags) {
     return syscall(SYS_memfd_create, name, flags);
-}
+} */
 
 /* memfd_create(2) flags */
 
