Index: src/modules/module-alsa-sink.c
===================================================================
--- src/modules/module-alsa-sink.c	(revision 1430)
+++ src/modules/module-alsa-sink.c	(revision 1432)
@@ -415,6 +419,9 @@
         goto fail;
     }
 
+    /* ALSA might tweak the sample spec, so recalculate the frame size */
+    frame_size = pa_frame_size(&ss);
+
     if (ss.channels != map.channels)
         /* Seems ALSA didn't like the channel number, so let's fix the channel map */
         pa_channel_map_init_auto(&map, ss.channels, PA_CHANNEL_MAP_ALSA);
