diff -up pulseaudio-8.0/man/pulse-daemon.conf.5.xml.in.disable_flat_volumes pulseaudio-8.0/man/pulse-daemon.conf.5.xml.in
--- pulseaudio-8.0/man/pulse-daemon.conf.5.xml.in.disable_flat_volumes	2015-12-31 00:28:35.000000000 -0600
+++ pulseaudio-8.0/man/pulse-daemon.conf.5.xml.in	2016-03-05 09:04:37.041037742 -0600
@@ -215,7 +215,7 @@ License along with PulseAudio; if not, s
       <p><opt>flat-volumes=</opt> Enable 'flat' volumes, i.e. where
       possible let the sink volume equal the maximum of the volumes of
       the inputs connected to it. Takes a boolean argument, defaults
-      to <opt>yes</opt>.</p>
+      to <opt>no</opt>.</p>
     </option>
 
   </section>
diff -up pulseaudio-8.0/src/daemon/daemon.conf.in.disable_flat_volumes pulseaudio-8.0/src/daemon/daemon.conf.in
--- pulseaudio-8.0/src/daemon/daemon.conf.in.disable_flat_volumes	2015-09-09 23:51:41.000000000 -0500
+++ pulseaudio-8.0/src/daemon/daemon.conf.in	2016-03-05 09:04:37.041037742 -0600
@@ -57,7 +57,7 @@ ifelse(@HAVE_DBUS@, 1, [dnl
 ; enable-lfe-remixing = yes
 ; lfe-crossover-freq = 120
 
-; flat-volumes = yes
+; flat-volumes = no 
 
 ifelse(@HAVE_SYS_RESOURCE_H@, 1, [dnl
 ; rlimit-fsize = -1
diff -up pulseaudio-8.0/src/pulsecore/core.c.disable_flat_volumes pulseaudio-8.0/src/pulsecore/core.c
--- pulseaudio-8.0/src/pulsecore/core.c.disable_flat_volumes	2015-12-31 00:28:35.000000000 -0600
+++ pulseaudio-8.0/src/pulsecore/core.c	2016-03-05 09:04:37.041037742 -0600
@@ -136,7 +136,7 @@ pa_core* pa_core_new(pa_mainloop_api *m,
     c->exit_idle_time = -1;
     c->scache_idle_time = 20;
 
-    c->flat_volumes = true;
+    c->flat_volumes = false;
     c->disallow_module_loading = false;
     c->disallow_exit = false;
     c->running_as_daemon = false;
