--- pulseaudio-0.9.6/src/Makefile.in.orig	2007-05-27 22:59:32.000000000 +0200
+++ pulseaudio-0.9.6/src/Makefile.in	2007-05-29 11:30:19.000000000 +0200
@@ -5028,7 +5028,7 @@
 
 suid: pulseaudio .libs/lt-pulseaudio
 	chown root $^
-	chmod u+s $^
+#	chmod u+s $^
 
 esdcompat: daemon/esdcompat.in Makefile
 	sed -e 's,@PACKAGE_VERSION\@,$(PACKAGE_VERSION),g' \
@@ -5052,7 +5052,7 @@
 
 install-exec-hook:
 	chown root $(DESTDIR)$(bindir)/pulseaudio ; true
-	chmod u+s $(DESTDIR)$(bindir)/pulseaudio
+#	chmod u+s $(DESTDIR)$(bindir)/pulseaudio
 	ln -sf pacat $(DESTDIR)$(bindir)/parec
 	rm -f $(DESTDIR)$(modlibexecdir)/*.a
 	rm -f $(DESTDIR)$(libdir)/libpulsedsp.a
