--- src/modules/bluetooth/sbc/sbc_primitives_mmx.c.orig	2012-02-28 21:32:40.167327239 -0600
+++ src/modules/bluetooth/sbc/sbc_primitives_mmx.c	2012-02-28 21:33:01.866085039 -0600
@@ -318,7 +318,7 @@
 				"movl          %k0, 4(%3)\n"
 			: "+r" (blk)
 			: "r" (&sb_sample_f[0][ch][sb]),
-				"i" ((char *) &sb_sample_f[1][0][0] -
+				"r" ((char *) &sb_sample_f[1][0][0] -
 					(char *) &sb_sample_f[0][0][0]),
 				"r" (&scale_factor[ch][sb]),
 				"r" (&consts),
