diff -up pulseaudio-11.1/src/daemon/daemon-conf.c.idletime pulseaudio-11.1/src/daemon/daemon-conf.c
--- pulseaudio-11.1/src/daemon/daemon-conf.c.idletime	2018-01-08 11:01:12.876616847 -0600
+++ pulseaudio-11.1/src/daemon/daemon-conf.c	2018-01-08 11:06:33.847102687 -0600
@@ -69,7 +69,7 @@ static const pa_daemon_conf default_conf
     .disallow_module_loading = false,
     .disallow_exit = false,
     .flat_volumes = false,
-    .exit_idle_time = 20,
+    .exit_idle_time = 4,
     .scache_idle_time = 20,
     .script_commands = NULL,
     .dl_search_path = NULL,
diff -up pulseaudio-11.1/src/daemon/daemon.conf.in.idletime pulseaudio-11.1/src/daemon/daemon.conf.in
--- pulseaudio-11.1/src/daemon/daemon.conf.in.idletime	2018-01-08 11:01:12.876616847 -0600
+++ pulseaudio-11.1/src/daemon/daemon.conf.in	2018-01-08 11:04:52.309999961 -0600
@@ -39,7 +39,7 @@ ifelse(@HAVE_DBUS@, 1, [dnl
 ; realtime-scheduling = yes
 ; realtime-priority = 5
 
-; exit-idle-time = 20
+; exit-idle-time = 4 
 ; scache-idle-time = 20
 
 ; dl-search-path = (depends on architecture)
