Index: /trunk/src/pulsecore/core-util.c
===================================================================
--- /trunk/src/pulsecore/core-util.c (revision 1418)
+++ /trunk/src/pulsecore/core-util.c (revision 1423)
@@ -185,5 +185,5 @@
     }
 #else
-	pa_log_warn("secure directory creation not supported on Win32.");
+    pa_log_warn("secure directory creation not supported on Win32.");
 #endif
 
@@ -954,5 +954,6 @@
 #endif
 
-            if ((f = fopen(fn, mode)) || errno != ENOENT) {
+            f = fopen(fn, mode);
+            if (f != NULL) {
                 if (result)
                     *result = pa_xstrdup(fn);
@@ -961,4 +962,9 @@
             }
 
+            if (errno != ENOENT) {
+                pa_log_warn("WARNING: failed to open configuration file '%s': %s",
+                    lfn, pa_cstrerror(errno));
+            }
+
             pa_xfree(lfn);
         }

